src/compiler/transformers/es2015.ts

```diff
@@ -2104,13 +2104,12 @@ namespace ts {
                 setTextRange(declarationList, node);
                 setCommentRange(declarationList, node);
 
-                if (node.transformFlags & TransformFlags.ContainsBindingPattern
-                    && (isBindingPattern(node.declarations[0].name)
-                        || isBindingPattern(lastOrUndefined(node.declarations).name))) {
+                const firstDeclaration = firstOrUndefined(declarations);

```

Comment:
Why look up the declarations before checking if there is work to do? If we need to ensure `firstDeclaration` is not `undefined`, add another `if` inside of the existing block.

Also, you shouldn't need to check _both_ `firstDeclaration` and `lastDeclaration` as both will be undefined if `declarations.length === 0` and both will have values if `declarations.length > 0`.